Mathlib/Topology/Algebra/Module/PointwiseConvergence.lean

```diff
@@ -0,0 +1,155 @@
+
+
+
+## References
+
+* [F. Bar, *Quuxes*][bibkey]
+
+## Tags
+
+Foobars, barfoos
+-/
+
+/-! ### Topology of pointwise convergence -/
+
+variable {α ι : Type*} [TopologicalSpace α]
+variable {𝕜 𝕜₁ 𝕜₂ : Type*} [NormedField 𝕜] [NormedField 𝕜₁] [NormedField 𝕜₂]
+variable {σ : 𝕜₁ →+* 𝕜₂}
+variable {E F : Type*} [AddCommGroup E] [TopologicalSpace E]
+  [AddCommGroup F] [TopologicalSpace F] [TopologicalAddGroup F]
+  [Module 𝕜 E] [Module 𝕜 F] [Module 𝕜₁ E] [Module 𝕜₂ F]
+
+open Topology
+
+variable (σ E F) in
+/-- The space of continuous linear maps equipped with the topology of pointwise/simple convergence,
+sometimes also called the strong operator topology.
+
+This topology is also known as the weak*-topology in the case that `σ = RingHom.id 𝕜` and `F = 𝕜` -/
+@[reducible]
+def PointwiseConvergenceCLM :=
+    UniformConvergenceCLM σ F {s : Set E | Finite s}
+
+@[inherit_doc]
+notation:25 E " →SLₛ[" σ "] " F => PointwiseConvergenceCLM σ E F
+
+@[inherit_doc]
+notation:25 E " →Lₛ[" R "] " F => PointwiseConvergenceCLM (RingHom.id R) E F
+
+namespace PointwiseConvergenceCLM
+
+protected theorem hasBasis_nhds_zero_of_basis
+    {ι : Type*} {p : ι → Prop} {b : ι → Set F} (h : (𝓝 0 : Filter F).HasBasis p b) :
+    (𝓝 (0 : E →SLₛ[σ] F)).HasBasis (fun Si : Set E × ι => Finite Si.1 ∧ p Si.2)
+      fun Si => { f : E →SLₛ[σ] F | ∀ x ∈ Si.1, f x ∈ b Si.2 } :=
+  UniformConvergenceCLM.hasBasis_nhds_zero_of_basis σ F { S | Finite S }
+    ⟨∅, Set.finite_empty⟩ (directedOn_of_sup_mem fun _ _ => Set.Finite.union) h
+
+protected theorem hasBasis_nhds_zero [TopologicalSpace F] [TopologicalAddGroup F] :
+    (𝓝 (0 : E →SLₛ[σ] F)).HasBasis
+      (fun SV : Set E × Set F => Finite SV.1 ∧ SV.2 ∈ (𝓝 0 : Filter F))
+      fun SV => { f : E →SLₛ[σ] F | ∀ x ∈ SV.1, f x ∈ SV.2 } :=
+  PointwiseConvergenceCLM.hasBasis_nhds_zero_of_basis (𝓝 0).basis_sets
+
+/-- In the topology of pointwise convergence, `a` converges to `a₀` iff for every `x : E`
+`a · x` converges to `a₀ x`. -/
+theorem tendsto_iff_forall_tendsto {p : Filter ι} {a : ι → E →SLₛ[σ] F} {a₀ : E →SLₛ[σ] F} :
+    Filter.Tendsto a p (𝓝 a₀) ↔ ∀ x : E, Filter.Tendsto (a · x) p (𝓝 (a₀ x)) := by
+  let _ := TopologicalAddGroup.toUniformSpace F
+  have _ : UniformAddGroup F := comm_topologicalAddGroup_is_uniform
+  suffices h : Filter.Tendsto a p (𝓝 a₀) ↔ ∀ x, TendstoUniformlyOn (a · ·) a₀ p {x} by
+    rw [h, forall_congr]
+    intro
+    rw [tendstoUniformlyOn_singleton_iff_tendsto]
+  rw [UniformConvergenceCLM.tendsto_iff_tendstoUniformlyOn]
+  unfold TendstoUniformlyOn
+  simp only [Set.mem_setOf_eq, Set.mem_singleton_iff, forall_eq]
+  constructor
+  · intro h x u hu
+    simpa using h {x} (Set.finite_singleton _) u hu
+  · intro h s hs u hu
+    rw [Filter.eventually_all_finite hs]
+    intro x _hx
+    exact h x u hu
+
+variable [ContinuousConstSMul 𝕜₂ F]
+
+variable (σ E F) in
+/-- Coercion from `E →Lₛ[𝕜] F` to `E →ₗ[𝕜] F` as a `𝕜`-linear map. -/
+def coeLM : (E →SLₛ[σ] F) →ₗ[𝕜₂] E →ₛₗ[σ] F where
```

Comment:
Is there precedence for using `LM` like this (without `C`)? I think this should probably just be called `coeLinearMap` or `toLinearMap`.